packages/core/src/BaseServiceClient.js

```diff
@@ -59,7 +59,14 @@ class BaseServiceClient {
    * @returns {Promise<ChannelStateReply>}
    */
   async getChannelState(channelId) {
-    const signatureBytes = await this._account.signData({ t: 'uint256', v: channelId });
+    const currentBlockNumber = await this._web3.eth.getBlockNumber();
+    const channelIdStr = new BigNumber(channelId).toFixed();
+    const signatureBytes = await this._account.signData(
+      { t: 'string', v: '__get_channel_state' },
+      { t: 'address', v: this._mpeContract.address },
+      { t: 'uint256', v: channelIdStr },
+      { t: 'uint256', v: currentBlockNumber },
+    );
 
```

Comment:
@vsbogd the signature format has been changed at the daemon end. This is the change we need to do from the consumer perspective. Rest of the changes are just refactoring.

Reply:
Ah, ok, as far as I see it is contract between daemon and client not Ethereum contract. Thanks.

Reply:
You are right. I think I used contract ambiguously. Sorry for that.